aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -318,19 +319,34 @@ Tensor imag(const Tensor& self) {
   }
 }
 
-Tensor& conj_out(const Tensor& self, Tensor& result) {
-  return unary_op_impl_out(result, self, conj_stub);
+Tensor resolve_conj(const Tensor& self) {
+  if (!self.is_conj()) { return self; }
+  auto result = at::empty_like(self, self.options());
+  result.set_conj(false);
+  // conjugation is handled in `copy_()`
+  return result.copy_(self);
 }
 
-Tensor _conj(const Tensor& self) { return unary_op_impl(self, at::conj_out); }
+Tensor _conj(const Tensor& self) {
+  Tensor self_;
+  auto impl = c10::make_intrusive<TensorImpl>(
+    Storage(self.storage()), self.key_set(), self.dtype());
+  impl->set_storage_offset(self.storage_offset());
+  impl->set_sizes_and_strides(self.sizes(), self.strides());
+  impl->set_conj(!self.is_conj());
+  self_ = Tensor(std::move(impl));
+  namedinference::propagate_names(self_, self);
+  return self_;
+}
 
 Tensor conj(const Tensor& self) {
-  if (!self.is_complex()) {
-    return self;
-  }
-  return at::_conj(self);
+  return self.conj();
```

Comment:
Blegh... I know what's going on here (manual cpp binding) but it still really looks like infinite regress to untrained eye. Maybe a brief comment here and a reference to https://github.com/pytorch/pytorch/issues/53622

Reply:
added a comment